test/test_quantized_nn_mods.py

```diff
@@ -49,53 +166,80 @@ def test_conv_api(self, use_bias, per_channel, qengine):
-            scale, zero_point = 1.0 / 255, 128
-            stride = (1, 1)
-            i_padding = (0, 0)
-            dilation = (1, 1)
+            use_channelwise = False
 
-            X = torch.randn(N, iC, H, W, dtype=torch.float32)
-            qX = torch.quantize_per_tensor(X, scale=scale, zero_point=128, dtype=torch.quint8)
+        input_feature_map_size = (H, W)
+        kernel_size = (kernel_h, kernel_w)
+        stride = (stride_h, stride_w)
+        padding = (pad_h, pad_w)
+        dilation = (dilation, dilation)
 
-            w = torch.randn(oC, iC // g, kH, kW, dtype=torch.float32)
+        with override_quantized_engine(qengine):
+            qconv_fn = qF.conv2d
+            conv_fn = F.conv2d
+            self._test_conv_api_impl(
+                qconv_fn, conv_fn, batch_size, input_channels_per_group,
+                input_feature_map_size, output_channels_per_group, groups,
+                kernel_size, stride, padding, dilation, X_scale, X_zero_point,
+                W_scale, W_zero_point, Y_scale, Y_zero_point, use_bias,
+                use_channelwise)
 
-            if per_channel:
-                scale_tensor = torch.ones(oC, dtype=torch.double)
-                zero_point_tensor = torch.zeros(oC, dtype=torch.long)
-                for i in range(len(scale_tensor)):
-                    scale_tensor[i] = (i + 1.0) / 255.0
+    @no_deadline
+    @given(batch_size=st.integers(1, 3),
+           input_channels_per_group=st.sampled_from([2, 4, 5, 8, 16, 32]),
+           D=st.integers(4, 8),
+           H=st.integers(4, 8),
+           W=st.integers(4, 8),
+           output_channels_per_group=st.sampled_from([2, 4, 5, 8, 16, 32]),
+           groups=st.integers(1, 4),
+           kernel_d=st.integers(1, 4),
+           kernel_h=st.integers(1, 4),
+           kernel_w=st.integers(1, 4),
+           stride_d=st.integers(1, 2),
+           stride_h=st.integers(1, 2),
+           stride_w=st.integers(1, 2),
+           pad_d=st.integers(0, 2),
+           pad_h=st.integers(0, 2),
+           pad_w=st.integers(0, 2),
+           dilation=st.integers(1, 2),
+           X_scale=st.floats(1.2, 1.6),
+           X_zero_point=st.integers(0, 4),
+           W_scale=st.lists(st.floats(0.2, 1.6), min_size=1, max_size=2),
+           W_zero_point=st.lists(st.integers(-5, 5), min_size=1, max_size=2),
+           Y_scale=st.floats(4.2, 5.6),
+           Y_zero_point=st.integers(0, 4),
+           use_bias=st.booleans(),
+           use_channelwise=st.booleans(),
+           qengine=st.sampled_from(("fbgemm")))
+    def test_conv3d_api(
+        self, batch_size, input_channels_per_group, D, H, W,
+        output_channels_per_group, groups, kernel_d, kernel_h, kernel_w,
+        stride_d, stride_h, stride_w, pad_d, pad_h, pad_w, dilation, X_scale,
+        X_zero_point, W_scale, W_zero_point, Y_scale, Y_zero_point, use_bias,
+        use_channelwise, qengine,
+    ):
+        """Tests the correctness of the conv3d function.
```

Comment:
Maybe also add a comment here conv2d is only supported by FBGEMM, not QNNPACK backend?